stubs/composer.stub

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Composer\Package;
+
+/**
+ * @phpstan-type AutoloadRules array{psr-0?: array<string, string[]>, psr-4?: array<string, string[]>, classmap?: list<string>, files?: list<string>, exclude-from-classmap?: list<string>}
+ * @phpstan-type DevAutoloadRules array{psr-0?: array<string, string[]>, psr-4?: array<string, string[]>, classmap?: list<string>, files?: list<string>}
+ */
+interface PackageInterface
+{
+}
```

Comment:
how did you generate this file?

Reply:
Manually. It’s just to fix a PHPdoc type that’s incorrect in Composer’s source code and make PHPStan happy.

Reply:
Basically if you look at the original PackageInterface you’ll see the PHPDoc type is nearly the same, I think I just removed an `|null` or similar.

Reply:
A stub is better than baselining the error, because the ignored error could suppress other legitimate errors.

Since it was only correcting one type, I removed the stub file and just directly set it in `AutoloadPlugin.php`

Reply:
with phpstan2 you can baseline the single type of error, not even the whole line ;-)

Reply:
no no i didn't want to convince you for the baseline, it was just a comment about phpstan2. also, you know that phpstan itself runs on level 8 and Ondrej himself told he doesn't want to raise it heheheheh

Reply:
I misunderstood. 😃

This particular error happens at level 7, but anyway getting to level 10 was really no problem since this repo is nowhere near the size of PHPStan itself. Plus composer has it's own set of rules specifically for extensions so it's nice to check those too.